packages/beacon-node/src/chain/prepareNextSlot.ts

```diff
@@ -120,6 +120,14 @@ export class PrepareNextSlotScheduler {
         const proposerIndex = prepareState.epochCtx.getBeaconProposer(prepareSlot);
         const feeRecipient = this.chain.beaconProposerCache.get(proposerIndex);
         if (feeRecipient) {
+          // Update if the builder flow can be used with this proposer, it might lead to
+          // an api call to the builder to check its status, so we will not await for
+          // its completion here as we want to trigger local execution here without
+          // delay
+          this.chain.updateBuilderStatus(clockSlot).catch((e) => {
```

Comment:
Without a more in-depth review this looks problematic. Since this is async there's a racing condition on producing the block and updating the status. Could be that the builder is disabled too late and the block is already produced?

Why not check at the end of every epoch (or some point, say at 2nd slot) the status of the previous epoch and update the status of the builder with it?
- That allows to have constant metrics of status
- You can subscribe to block event instead of `getSlotsPresent()` and have real-time cumulative status of blocks per epoch, which can also go into a metric

Reply:
:+1:  for counting the slots via event and the corresponding metric

However with regard to calling `updateBuilderStatus` here: for `disabling`, this is synchronous because the builder's check status api is only being called for `enabling`, which if it doesn't respond within a second should anyway be ok to have builder disabled for the next slot's block.

I think its a good idea to check for builder status  here before proposing next slot.

Reply:
> However with regard to calling updateBuilderStatus here: for disabling, this is synchronous because the builder's check status api is only being called for enabling

Got it, then can you find a way to make it more obvious on the code that disabling is sync?

Reply:
done